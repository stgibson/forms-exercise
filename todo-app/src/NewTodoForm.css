.NewTodoForm {
  margin-bottom: 20px;
}

.NewTodoForm-input {
  margin-left: 5px;
  margin-right: 10px;
}