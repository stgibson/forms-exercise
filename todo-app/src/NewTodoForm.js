import React, { useState } from "react";
import "./NewTodoForm.css";

const NewTodoForm = ({ addTodo }) => {
  const initFormData = { task: "" };
  const [formData, setFormData] = useState(initFormData);

  const handleChange = (event) => {
    const { name, value } = event.target;
    setFormData({ ...formData, [name]: value });
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    addTodo(formData.task);
    setFormData(initFormData);
  };

  return (
    <form className="NewTodoForm" onSubmit={ handleSubmit }>
      <label htmlFor="task">Task:</label>
      <input
        id="task"
        className="NewTodoForm-input"
        name="task"
        type="text"
        onChange={ handleChange }
        value={ formData.task }
      />
      <button type="submit">Add Todo</button>
    </form>
  );
};

export default NewTodoForm;