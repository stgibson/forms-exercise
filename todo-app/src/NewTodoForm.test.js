import React from "react";
import NewTodoForm from "./NewTodoForm";
import { render } from "@testing-library/react";

it("renders without crashing", () => {
  render(<NewTodoForm />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});