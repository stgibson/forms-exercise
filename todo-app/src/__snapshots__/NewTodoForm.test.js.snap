// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewTodoForm"
  >
    <label
      for="task"
    >
      Task:
    </label>
    <input
      class="NewTodoForm-input"
      id="task"
      name="task"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
